pkg/pquerier/pqueriertest/dataframe_query_integration_test.go

```diff
@@ -959,3 +959,353 @@ func (suite *testSelectDataframeSuite) TestSparseStringAndNumericColumnsDatafram
 		}
 	}
 }
+
+func (suite *testSelectDataframeSuite) TestSparseNumericColumnsWithEmptyColumnsDataframe() {
+	labelSetLinux := utils.LabelsFromStringList("os", "linux")
+	labelSetWindows := utils.LabelsFromStringList("os", "windows")
+	expectedTimeColumn := []int64{
+		suite.basicQueryTime,
+		suite.basicQueryTime + tsdbtest.MinuteInMillis,
+		suite.basicQueryTime + 2*tsdbtest.MinuteInMillis,
+		suite.basicQueryTime + 3*tsdbtest.MinuteInMillis,
+		suite.basicQueryTime + 4*tsdbtest.MinuteInMillis}
+	expectedColumns := map[string][]interface{}{
```

Comment:
I would use a composite key of `metric-label` something like `cpu_0-linux`, `cpu_1-linux`, `cpu_1-win`...
it's a lot clearer then comparing two DFs with 5 rows each against a 10 items list..

Reply:
you're welcome to refactor this in separate PR